lib/model/model.go

```diff
@@ -1567,12 +1568,15 @@ func sendIndexTo(minSequence int64, conn protocol.Connection, folder string, fs
 }
 
 func (m *Model) updateLocalsFromScanning(folder string, fs []protocol.FileInfo) {
-	m.updateLocals(folder, fs)
-
 	m.fmut.RLock()
 	folderCfg := m.folderCfgs[folder]
+	runner, _ := m.folderRunners[folder]
 	m.fmut.RUnlock()
 
+	// Validate all local changes and update the database
+	fs = runner.validateAndUpdateLocalChanges(fs)
```

Comment:
I did this, because I need to a `IndexUpdated()` to force the modified files being downloaded again after the database gets updated. I wanted to avoid to executions of `updateLocals`.